neural_compressor/torch/algorithms/weight_only/save_load.py

```diff
@@ -37,14 +38,516 @@ def save(model, output_dir="./saved_results"):
 
     # MethodType 'save' not in state_dict
     del model.save
-    torch.save(model, qmodel_file_path)
+    torch.save(model.state_dict(), qmodel_weight_file_path)
 
-    logger.info("Save quantized model to {}.".format(qmodel_file_path))
+    logger.info("Save quantized model weight to {}.".format(qmodel_weight_file_path))
     logger.info("Save configuration of quantized model to {}.".format(qconfig_file_path))
 
 
-def load(output_dir="./saved_results"):
-    qmodel_file_path = os.path.join(os.path.abspath(os.path.expanduser(output_dir)), WEIGHT_NAME)
-    model = torch.load(qmodel_file_path)
-    logger.info("Quantized model loading successful.")
+def load(model_name_or_path, original_model=None, format=LoadFormat.DEFAULT, device="cpu", *model_args, **kwargs):
+    """Load quantized weight-only quantization model.
+
+    1. Load INC weight-only quantized model in local.
+    2. Load HuggingFace weight-only quantized model,
+       including GPTQ/AWQ models and upstreamed INC quantized models in HF model hub.
+
+    Args:
+        model_name_or_path (str):  torch checkpoint directory or hugginface model_name_or_path.
+            If 'format' is set to 'huggingface', it means the huggingface model_name_or_path.
+            If 'format' is set to 'default', it means the 'checkpoint_dir'.
+            Parameter should not be None. it coworks with 'original_model' parameter to load INC
+            weight-only quantized model in local.
+        original_model (torch.nn.module, optional): original model before quantization.
+            Needed if 'format' is set to 'default' and not TorchScript model.Defaults to None.
+        format (str, optional): 'defult' for loading INC weight-only quantized model.
+            'huggingface' for loading huggingface WOQ causal language model. Defaults to "default".
+        model_args (sequence of positional arguments, optional):
+            all remaining positional arguments for loading huggingface models.
+            will be passed to the huggingface model's `__init__` method.
+        kwargs (remaining dictionary of keyword arguments, optional):
+            remaining dictionary of keyword arguments for loading huggingface models.
+            will be passed to the huggingface model's `__init__` method, such as 'trust_remote_code', 'revision'.
+    Returns:
+        torch.nn.Module: quantized model
+    """
+    if format == LoadFormat.HUGGINGFACE:
+        model = _load_hf_woq_model(model_name_or_path, *model_args, **kwargs)
```

Comment:
The loaded model is still INC provided model, how about using `_load_hf_format_model` and `_load_inc_format_model`?

Reply:
updated in https://github.com/intel/neural-compressor/pull/1826/commits/939a261dd4554ea6e45dc2ae23586ab7eda4ed68